### DISCUSSION TITLE
How can I create a custom alert in TCA?

### DISCUSSION THREAD

**Discussion by hossamsherif-reo** [pointfreeco/swift-composable-architecture#1049] [Q&A] [answered]
So I am trying to create a custom alert that has different design from the SwiftUI.Alert that is currently supported by TCA with `AlertState<Action>` and the `.alert` modifier, so how can I reach a similar behavior but with custom alert UI? 
Thanks 🙏

**Comment by stephencelis** [comment, marked as answer]
Hi @hossamsherif-reo! In isowords we have an example "bottom menu" component that might be a good reference:

https://github.com/pointfreeco/isowords/tree/b87f4c61c0523564adf17c031f9fa4b88a40d631/Sources/BottomMenu

The module includes 2 files. The first, "BottomMenu.swift," includes just the core SwiftUI view, that can be configured with `Text`s, button action closures, and is driven by an optional binding of `BottomMenu` state:

https://github.com/pointfreeco/isowords/blob/b87f4c61c0523564adf17c031f9fa4b88a40d631/Sources/BottomMenu/BottomMenu.swift

The second file, "ComposableBottomMenu.swift," contains all the glue needed to drive the component UI from TCA:

https://github.com/pointfreeco/isowords/blob/b87f4c61c0523564adf17c031f9fa4b88a40d631/Sources/BottomMenu/ComposableBottomMenu.swift

And this view and glue is used in our "GameCore" module, where `BottomMenuState` is held in `GameState`:

https://github.com/pointfreeco/isowords/blob/a4cc82c686cab65203d00ad0b207b8ae2acc45a4/Sources/GameCore/GameCore.swift#L35

Hope this helps get you started!

**Reply by hossamsherif-reo** [reply]
Hello @stephencelis! That's awesome! Yes this would be a great starting point. Thanks a lot 🙏
